middleware/index.ts

```diff
@@ -184,13 +200,29 @@ export default function applySfFxMiddleware(request: any, state: any, resultArgs
         throw new Error('Request Data not provided');
     }
 
+    // Logger is the last element in the array
+    const logger: Logger = resultArgs[resultArgs.length-1];
+    if (!logger) {
+        throw new Error('Logger not provided in resultArgs from node system function');
+    }
+
+    //use secret her in lieu of DEBUG runtime environment var until we haave deployment time support of config var
+    const debugSecret = getSecret("sf-debug", "DEBUG", logger);
+    logger.info(`DEBUG flag is ${debugSecret}`);
+    if(debugSecret || LoggerLevel.DEBUG === logger.getLevel() || process.env.DEBUG) {
+        //for dev preview, we log the ENTIRE raw request, may need to filter sensitive properties out later
+        //the hard part of filtering to know which name to filter
+        let childLogger: Logger = logger.child('raw_request', request);
+        childLogger.info('debug raw request in middleware');
+    }
```

Comment:
Got test coverage for `getSecret` and above?

Reply:
will add for above... for getSecret, will replace with sdk api.. so no plan to test that.